src/compiler/program.ts

```diff
@@ -4029,7 +4034,7 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             for (let index = 0; index < moduleNames.length; index++) {
                 const resolution = resolutions[index].resolvedModule;
                 const moduleName = moduleNames[index].text;
-                const mode = getModeForUsageLocation(file, moduleNames[index]);
+                const mode = getModeForUsageLocation(file, moduleNames[index], optionsForFile);

```

Comment:
This should always be `getRedirectReferenceForResolution(file)?.commandline.options || options`

since optionsForFile are to handle "allowJs" etc